projects/Mallard/src/hooks/use-issue-summary.tsx

```diff
@@ -55,21 +55,29 @@ const IssueSummaryProvider = ({ children }: { children: React.ReactNode }) => {
             .catch(e => {
                 setError(e.message)
             })
-    }
 
     useEffect(() => {
         NetInfo.addEventListener(({ isConnected }) => {
+            // try and get a fresh summary until we made it to online
             if (!hasConnected.current) {
                 hasConnected.current = isConnected
 
                 grabIssueSummary(isConnected)
             }
+        })
 
-            AppState.addEventListener('change', appState => {
-                if (appState === 'active') {
+        AppState.addEventListener('change', async appState => {
```

Comment:
Previously, every time we changed connection we would add a new event listener for the `AppState` `change` event. Have moved this out and so that it only gets called once and derives `isConnected` itself.

Reply:
So all this stuffs happens separately for any code calling `useIssueSummary`? Ie. it's going to try fetching twice in parallel if we have two `useIssueSummary`?

Reply:
If you mean what I think you mean, this is happening inside a context provider, so this only happens in the provider, not in the component using the hook. The hook just receives the state as it is in the provider. Unless that's not what you mean ...